src/train.py

```diff
@@ -40,7 +58,13 @@ def train(args):
 
             character, mel, mel_input, pos_text, pos_mel, _ = data
 
-    raise Exception("TODO: Implement")
+        for batch in dataloader:
+            # choose loss function here!
+            model.decode(model.encode(batch))
+        evaluate(model, valid_dataset)
```

Comment:
Sorry if I missed it, but is this function defined anywhere? If not can you comment it and add TODOs as needed?

Reply:
Yeah it is not defined yet, but this would be where we evaluate on the dev set to monitor performance during training.  It should be a stub rn